### ISSUE TITLE
User Role Page get error when choosing next page

### ISSUE THREAD

**Issue by huyhoangttql** [magento/magento2#36896]
### Preconditions and environment

- Magento version 2.4.5
- Exception message:
1 exception(s):
Exception #0 (Exception): Warning: Array to string conversion in /home/mage/public_html/vendor/magento/framework/Escaper.php on line 309

Exception #0 (Exception): Warning: Array to string conversion in /home/mage/public_html/vendor/magento/framework/Escaper.php on line 309
<pre>#1 Magento\Framework\Escaper->escapeJs() called at [vendor/magento/framework/View/Element/AbstractBlock.php:899]
#2 Magento\Framework\View\Element\AbstractBlock->escapeJs() called at [vendor/magento/module-user/Block/Role/Grid/User.php:209]
#3 Magento\User\Block\Role\Grid\User->getUsers() called at [vendor/magento/module-user/Block/Role/Grid/User.php:140]
#4 Magento\User\Block\Role\Grid\User->_prepareColumns() called at [vendor/magento/module-backend/Block/Widget/Grid/Extended.php:500]
#5 Magento\Backend\Block\Widget\Grid\Extended->_prepareGrid() called at [vendor/magento/module-backend/Block/Widget/Grid.php:478]
#6 Magento\Backend\Block\Widget\Grid->_beforeToHtml() called at [vendor/magento/framework/View/Element/AbstractBlock.php:1094]
#7 Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}() called at [vendor/magento/framework/View/Element/AbstractBlock.php:1099]
#8 Magento\Framework\View\Element\AbstractBlock->_loadCache() called at [vendor/magento/framework/View/Element/AbstractBlock.php:660]
#9 Magento\Framework\View\Element\AbstractBlock->toHtml() called at [vendor/magento/framework/View/Layout.php:578]
#10 Magento\Framework\View\Layout->_renderBlock() called at [vendor/magento/framework/View/Layout.php:555]
#11 Magento\Framework\View\Layout->renderNonCachedElement() called at [generated/code/Magento/Framework/View/Layout/Interceptor.php:149]
#12 Magento\Framework\View\Layout\Interceptor->renderNonCachedElement() called at [vendor/magento/framework/View/Layout.php:510]
#13 Magento\Framework\View\Layout->renderElement() called at [generated/code/Magento/Framework/View/Layout/Interceptor.php:140]
#14 Magento\Framework\View\Layout\Interceptor->renderElement() called at [vendor/magento/framework/View/Layout.php:606]
#15 Magento\Framework\View\Layout->_renderContainer() called at [vendor/magento/framework/View/Layout.php:557]
#16 Magento\Framework\View\Layout->renderNonCachedElement() called at [generated/code/Magento/Framework/View/Layout/Interceptor.php:149]
#17 Magento\Framework\View\Layout\Interceptor->renderNonCachedElement() called at [vendor/magento/framework/View/Layout.php:510]
#18 Magento\Framework\View\Layout->renderElement() called at [generated/code/Magento/Framework/View/Layout/Interceptor.php:140]
#19 Magento\Framework\View\Layout\Interceptor->renderElement() called at [vendor/magento/framework/View/Layout.php:975]
#20 Magento\Framework\View\Layout->getOutput() called at [generated/code/Magento/Framework/View/Layout/Interceptor.php:347]
#21 Magento\Framework\View\Layout\Interceptor->getOutput() called at [vendor/magento/framework/View/Result/Page.php:260]
#22 Magento\Framework\View\Result\Page->render() called at [vendor/magento/framework/View/Result/Layout.php:171]
#23 Magento\Framework\View\Result\Layout->renderResult() called at [generated/code/Magento/Backend/Model/View/Result/Page/Interceptor.php:140]
#24 Magento\Backend\Model\View\Result\Page\Interceptor->renderResult() called at [vendor/magento/framework/App/Http.php:120]
#25 Magento\Framework\App\Http->launch() called at [generated/code/Magento/Framework/App/Http/Interceptor.php:23]
#26 Magento\Framework\App\Http\Interceptor->launch() called at [vendor/magento/framework/App/Bootstrap.php:264]
#27 Magento\Framework\App\Bootstrap->run() called at [pub/index.php:30]
</pre>

### Steps to reproduce

1. Create more than 20 admin users with the same role
2. Choose 20 per page in Role user grid
3. Go to System > User Roles > Choose a role > Role Users tab > Tick on one user role > Click next page

### Expected result

Show data in next page

### Actual result

Redirect to Dashboard

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by huyhoangttql** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hi @huyhoangttql,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the attached video.


**Steps to reproduce**


- Create more than 20 admin users with the same role
- Choose 20 per page in Role user grid
- Go to System > User Roles > Choose a role > Role Users tab > Tick on one user role > Click next page

![Users-Permissions-System-Magento-Admin](https://user-images.githubusercontent.com/51680745/220340922-e6bfd874-c63d-416a-a2f8-fd5302c4136b.png)




https://user-images.githubusercontent.com/51680745/220340411-65333690-bf23-4404-a30d-50f7ea631bf4.mov


We are able to get the data for the Next page.


Kindly recheck the behavior on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.


Thanks.

**Comment by creativeWebSoft** [comment]
Try my patch:
```
diff --git a/vendor/magento/module-user/Block/Role/Grid/User.php b/vendor/magento/module-user/Block/Role/Grid/User.php
index b1180e6..295f17b 100644
--- a/vendor/magento/module-user/Block/Role/Grid/User.php
+++ b/vendor/magento/module-user/Block/Role/Grid/User.php
@@ -137,7 +137,7 @@
                 'header_css_class' => 'a-center',
                 'type' => 'checkbox',
                 'name' => 'in_role_users',
-                'values' => $this->getUsers(),
+                'values' => $this->getUsers(true),
                 'align' => 'center',
                 'index' => 'user_id'
             ]
```
Magento 2.4.4 with php8.1

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
@creativeWebSoft Thanks for your contribution & collaboration over here.

@huyhoangttql Kindly recheck the issue on Magento 2.4-develop instance and check if any 3rd party extensions / modules enabled is causing this issue.

Kindly refer the attched video: 


https://user-images.githubusercontent.com/51680745/223106773-b0a137dd-2c21-48a6-b8a5-2cc69f394027.mov



Please let us know if you are still facing any issue.

Thanks.

**Comment by engcom-Bravo** [comment]
Hi @huyhoangttql,

We have noticed that this issue has not been updated for a period of 14 Days. Hence we assume that this issue is fixed now, so we are closing it. Please raise a fresh ticket or reopen this ticket if you need more assistance on this.

Thanks.

**Comment by Tomasz-Silpion** [comment]
The problem was reported on Magento version 2.4.5 while @engcom-Bravo stated that the issue is not reproducible on Magento 2.4-develop. Indeed, the fix was applied in 2.4.6 and merchants affected should either upgrade their stores or apply patch from https://github.com/magento/magento2/commit/f8a10cd18b5311984cbc4860056fa960b69a944e#diff-f0d9131306818be04d4a36e8c2943463ec28ea456b963995aa3ec1ee86840c35
